src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/StringCollectionTextWriter.cs

```diff
@@ -41,12 +42,12 @@ public override Encoding Encoding
         /// A collection of entries buffered by this instance of <see cref="StringCollectionTextWriter"/>.
         /// </summary>
         // internal for testing purposes.
-        internal BufferEntryCollection Buffer { get; }
+        internal BufferedHtmlContent Content { get; }
 
         /// <inheritdoc />
         public override void Write(char value)
         {
-            Buffer.Add(value.ToString());
+            Content.Append(value.ToString());
```

Comment:
It is a character array which in turn is converted to a string. Instead of char -> char[] -> string it is just char -> string here.